test/setup/cluster_setup.go

```diff
@@ -386,45 +387,59 @@ func (c *Cluster) createUsersCryptoMaterials() error {
 	}
 
 	for _, user := range c.Users {
-		pemUserCert, pemUserKey, err := testutils.IssueCertificate("Cluster User: "+user, "127.0.0.1", keyPair)
+		c.CreateOrUpdateUserCerts(user, keyPair)
+	}
 
-		userCertPath := path.Join(c.testDirAbsPath, "users", user+".pem")
-		pemUserCertFile, err := os.Create(userCertPath)
-		if err != nil {
-			return err
-		}
-		_, err = pemUserCertFile.Write(pemUserCert)
-		if err != nil {
-			return err
-		}
-		if err = pemUserCertFile.Close(); err != nil {
-			return err
-		}
+	return nil
+}
 
-		userKeyPath := path.Join(c.testDirAbsPath, "users", user+".key")
-		pemUserKeyFile, err := os.Create(userKeyPath)
+func (c *Cluster) CreateOrUpdateUserCerts(user string, keyPair tls.Certificate) error {
+	var err error
+	if reflect.ValueOf(keyPair).IsZero() {
+		keyPair, err = tls.X509KeyPair(c.rootCAPemCert, c.caPrivKey)
 		if err != nil {
 			return err
 		}
-		if _, err = pemUserKeyFile.Write(pemUserKey); err != nil {
-			return err
-		}
-		if err = pemUserKeyFile.Close(); err != nil {
-			return err
-		}
+	}
+
+	pemUserCert, pemUserKey, err := testutils.IssueCertificate("Cluster User: "+user, "127.0.0.1", keyPair)
+
+	userCertPath := path.Join(c.testDirAbsPath, "users", user+".pem")
+	pemUserCertFile, err := os.Create(userCertPath)
+	if err != nil {
+		return err
+	}
+	_, err = pemUserCertFile.Write(pemUserCert)
+	if err != nil {
+		return err
+	}
+	if err = pemUserCertFile.Close(); err != nil {
+		return err
+	}
+
+	userKeyPath := path.Join(c.testDirAbsPath, "users", user+".key")
+	pemUserKeyFile, err := os.Create(userKeyPath)
+	if err != nil {
+		return err
+	}
+	if _, err = pemUserKeyFile.Write(pemUserKey); err != nil {
+		return err
+	}
+	if err = pemUserKeyFile.Close(); err != nil {
+		return err
+	}
```

Comment:
you can use `os.WriteFile` t do all this in shorter way.

Reply:
Done